plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt

```diff
@@ -87,10 +87,7 @@ private fun YamlNode.toPod(): Pod =
             Pod(
                 name = name,
                 version = version,
-                dependencies = mapEntry.value.yamlList.items.map {
-                    val (depName, depVersion) = parseNameAndVersion(it.yamlScalar.content)
-                    Pod(depName, depVersion)
-                }
+                dependencies = mapEntry.value.yamlList.items.map { it.toPod() }
```

Comment:
Nit: As it's a bit hard to see from the collapsed diff, maybe explain in the commit message that this is calling `toPod()` recursively, assuming that all list entries are scalars.